airflow/www/views.py

```diff
@@ -2296,30 +2337,35 @@ def ordering(dag):
             stringify_timedelta=stringify_timedelta,
             get_human_readable_cron=cron.get_human_readable_cron,
             dagbag=dagbag,
-            log_regex=log_regex,
-            xcom_log_urls_key=conf.get('core', 'xcom_log_urls_key'),
-            get_xcom=functools.partial(get_xcom, session),
             include_owners=include_owners,
             owner_dict=owner_dict,
             auto_complete_data=auto_complete_data,
             use_coatue_ui=use_coatue_ui,
             all_dags=all_dags)
 
-def get_xcom(session, dag_id, task_id, dttm):
-    """
-    Given a specific run of a specific task, pulls all the XCom data from that run and returns a dict of the data.
+    @expose("/get_xcom")
+    @provide_session
+    def get_xcom(self, session=None):
+        """
+        Given a specific run of a specific task, pulls all the XCom data from that run and returns a dict of the data.
 
-    :param session: Session to query
-    :param dag_id: DAG ID
-    :param task_id: Task ID
-    :param dttm: Execution datetime of run
-    :return: Dict of key-value pairs from XCom
-    """
-    XCom = models.XCom
-    xcomlist = session.query(XCom).filter(
-        XCom.dag_id == dag_id, XCom.task_id == task_id,
-        XCom.execution_date == dttm).all()
-    return {xcom.key: xcom.value for xcom in xcomlist}
+        :param session: Session to query
+        :param dag_id: DAG ID
+        :param task_id: Task ID
+        :param dttm: Execution datetime of run
+        :return: Dict of key-value pairs from XCom
+        """
+        XCom = models.XCom
+        task_id = request.args.get("task_id")
+        dag_id = request.args.get("dag_id")
+        dttm = pendulum.parse(request.args.get("dttm"))
+        xcomlist = session.query(XCom).filter(
+            XCom.dag_id == dag_id, XCom.task_id == task_id,
+            XCom.execution_date == dttm).all()
+        payload = {}
```

Comment:
you can do set comprehension here btw e.g.:

```suggestion
        payload = {xcom.key: xcom.value for xcom in xcomlist}
```